src/components/MapView/Legends/index.tsx

```diff
@@ -78,21 +82,36 @@ function Legends({ classes, layers, extent }: LegendsProps) {
   const [open, setOpen] = useState(true);
   const isAnalysisLayerActive = useSelector(isAnalysisLayerActiveSelector);
   const analysisResult = useSelector(analysisResultSelector);
+  const { translatedColumns } = useAnalysisTableColumns(analysisResult);
   const features = analysisResult?.featureCollection.features;
   const hasData = features ? features.length > 0 : false;
 
   const { t } = useSafeTranslation();
 
-  const handleAnalysisDownload = (e: React.ChangeEvent<{}>): void => {
-    e.preventDefault();
-    downloadToFile(
-      {
-        content: JSON.stringify(features),
-        isUrl: false,
-      },
-      analysisResult?.getTitle() ?? 'prism_extract',
-      'application/json',
-    );
+  const handleAnalysisDownload = (format: 'csv' | 'geojson'): void => {
+    switch (format) {
+      case 'geojson':
+        downloadToFile(
+          {
+            content: JSON.stringify(features),
+            isUrl: false,
+          },
+          analysisResult?.getTitle() ?? 'prism_extract',
+          'application/json',
+        );
+        break;
+      case 'csv':
+        if (
+          analysisResult &&
+          (analysisResult instanceof BaselineLayerResult ||
+            analysisResult instanceof PolygonAnalysisResult)
```

Comment:
Since this is used twice, it might be worth extracting it into a variable with a clear name. Maybe `layerAcceptsCSVDownload` or something like that?

Reply:
It has been changed by the last commits